include/deal.II/matrix_free/operators.h

```diff
@@ -678,13 +686,18 @@ namespace MatrixFreeOperators
   void
   Base<dim,Number>::
   initialize (const MatrixFree<dim,Number> &data_,
-              const MGConstrainedDoFs &mg_constrained_dofs,
-              const unsigned int level)
+              const MGConstrainedDoFs      &mg_constrained_dofs,
+              const unsigned int            level)
   {
     AssertThrow (level != numbers::invalid_unsigned_int,
                  ExcMessage("level is not set"));
+    if (data_.n_macro_cells() > 0)
+      {
+        AssertDimension(static_cast<int>(level),
```

Comment:
essentially, it makes sure that the provided `data` object corresponds to the provided `level`? Are there cases when `data_.n_macro_cells()` is not yet positive? I guess not. If so, we may as well remove the `level `from the arguments and take it from this assertion.